### PR TITLE
Handle sub organization resource limit error

### PR DESCRIPTION
**Pull request by sadilchamishka**

### CODE REVIEW

File: apps/console/src/features/organizations/components/add-organization-modal.tsx [hunk 1]
```diff
@@ -23,15 +23,16 @@ import { Heading, LinkButton, Message, PrimaryButton } from "@wso2is/react-compo
...
-import { AppState, EventPublisher } from "../../core";
+import { AppState, EventPublisher, TierLimitReachErrorModal } from "../../core";
 import { addOrganization } from "../api";
 import {
     ORGANIZATION_DESCRIPTION_MAX_LENGTH,
     ORGANIZATION_DESCRIPTION_MIN_LENGTH,
     ORGANIZATION_NAME_MAX_LENGTH,
     ORGANIZATION_NAME_MIN_LENGTH,
-    ORGANIZATION_TYPE
+    ORGANIZATION_TYPE, OrganizationManagementConstants
```
Comment (kavinda1995) [inline apps/console/src/features/organizations/components/add-organization-modal.tsx:35]:
```suggestion
    ORGANIZATION_TYPE,
    OrganizationManagementConstants
```

File: apps/console/src/features/organizations/components/add-organization-modal.tsx [hunk 2]
```diff
@@ -68,15 +69,16 @@ export const AddOrganizationModal: FunctionComponent<AddOrganizationModalPropsIn
...
     const { t } = useTranslation();
 
-    const dispatch = useDispatch();
+    const dispatch: Dispatch<any> = useDispatch();
 
     const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
     const [ error, setError ] = useState<string>("");
 
-    const submitForm = useRef<() => void>();
+    const submitForm: any = useRef<() => void>();
```
Comment (kavinda1995) [inline apps/console/src/features/organizations/components/add-organization-modal.tsx:77]: Shall we not use `any`? We can type this as `RefObject<HTMLFormElement>`

File: apps/console/src/features/organizations/components/add-organization-modal.tsx [hunk 3]
```diff
@@ -68,15 +69,16 @@ export const AddOrganizationModal: FunctionComponent<AddOrganizationModalPropsIn
 
     const { t } = useTranslation();
 
-    const dispatch = useDispatch();
+    const dispatch: Dispatch<any> = useDispatch();
```
Comment (kavinda1995) [inline apps/console/src/features/organizations/components/add-organization-modal.tsx:72]:
```suggestion
    const dispatch: Dispatch = useDispatch();
```

Comment (kavinda1995) [reply apps/console/src/features/organizations/components/add-organization-modal.tsx:72]: Check this btw :D

File: apps/console/src/features/organizations/components/add-organization-modal.tsx [hunk 4]
```diff
@@ -68,15 +69,16 @@ export const AddOrganizationModal: FunctionComponent<AddOrganizationModalPropsIn
...
 
     const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
     const [ error, setError ] = useState<string>("");
 
-    const submitForm = useRef<() => void>();
+    const submitForm: any = useRef<() => void>();
 
     const eventPublisher: EventPublisher = EventPublisher.getInstance();
-    const currentOrganization = useSelector((state: AppState) => state.organization.organization);
+    const currentOrganization: any = useSelector((state: AppState) => state.organization.organization);
```
Comment (kavinda1995) [inline apps/console/src/features/organizations/components/add-organization-modal.tsx:80]: Let's refrain from using `any` as type

File: apps/console/src/features/organizations/components/add-organization-modal.tsx [hunk 5]
```diff
@@ -114,15 +116,19 @@ export const AddOrganizationModal: FunctionComponent<AddOrganizationModalPropsIn
                     onUpdate();
                 }
             })
-            .catch((error) => {
+            .catch((error: any) => {
```
Comment (kavinda1995) [inline apps/console/src/features/organizations/components/add-organization-modal.tsx:119]: Let's replace `any` with the correct type

File: apps/console/src/features/organizations/components/add-organization-modal.tsx (hunk 1 above)
Comment (sadilchamishka) [reply apps/console/src/features/organizations/components/add-organization-modal.tsx]: Fixed by - https://github.com/wso2/identity-apps/pull/3792/commits/913b79b7ab6c4605441b5d57363c7e722c0c1b1e

File: apps/console/src/features/organizations/components/add-organization-modal.tsx (hunk 3 above)
Comment (sadilchamishka) [reply apps/console/src/features/organizations/components/add-organization-modal.tsx]: Fixed by - https://github.com/wso2/identity-apps/pull/3792/commits/913b79b7ab6c4605441b5d57363c7e722c0c1b1e

File: apps/console/src/features/organizations/components/add-organization-modal.tsx (hunk 4 above)
Comment (sadilchamishka) [reply apps/console/src/features/organizations/components/add-organization-modal.tsx]: Fixed by - https://github.com/wso2/identity-apps/pull/3792/commits/913b79b7ab6c4605441b5d57363c7e722c0c1b1e#diff-1d62bd4a5e730a6f701d6278738212ce35873624667c0cf20721d90a7306290eR81

**Review by NipuniBhagya — APPROVED** [review]

**Pull request merged**